synapse/cortex.py

```diff
@@ -1478,19 +1478,28 @@ async def initServicePassive(self):
 
     async def initStormPool(self):
 
-        byts = self.slab.get(b'storm:pool', db='cell:conf')
-        if byts is None:
-            return
+        try:
 
-        url, opts = s_msgpack.un(byts)
+            byts = self.slab.get(b'storm:pool', db='cell:conf')
+            if byts is None:
+                return
 
-        self.stormpoolurl = url
-        self.stormpoolopts = opts
+            url, opts = s_msgpack.un(byts)
 
-        self.stormpool = await s_telepath.open(url)
+            self.stormpoolurl = url
+            self.stormpoolopts = opts
 
-        # make this one a fini weakref vs the fini() handler
-        self.onfini(self.stormpool)
+            async def onlink(proxy, urlinfo):
+                _url = s_urlhelp.sanitizeUrl(s_telepath.zipurl(urlinfo))
+                logger.debug(f'Stormpool client connected to {_url}')
+
+            self.stormpool = await s_telepath.open(url, onlink=onlink)
+
+            # make this one a fini weakref vs the fini() handler
+            self.onfini(self.stormpool)
+
+        except Exception as e:
+            logger.exception(f'Error starting stormpool, it will not be available: {e}')
```

Comment:
Is there a command or something to get it to retry? That might be worth mentioning in the error message. If not, maybe add that it's unavailable until the next restart?

Reply:
We don't have a retry command; and unless we inject the pool status in to something like `getCellInfo()` we don't have a great way to signal that is failed.

The obvious codepaths for this to fail / blowup would be:

1) lmdb read error
2) msgpack unpack error
3) clientV2 __anit__ calls chopurl(). We're now protecting against a chopurl failure in the `cortex.storm.pool.set` command.

Since we call this in-line of startup, we can't have this preventing Cortex startup for what is an optional piece of functionality.

Reply:
The new `telepath.open()` paradigm should make this nearly moot. It does not depend on any connection working to construct the ClientV2 anymore :)